src/components/menu/Menu.tsx

```diff
@@ -0,0 +1,237 @@
+import Box, { pickBoxProps } from '../box';
+import MenuItem from './MenuItem';
+import theme from './theme.css';
+import uiUtilities from '@teamleader/ui-utilities';
+import { BoxProps } from '../box/Box';
+import { GenericComponent } from '../../@types/types';
+
+export type POSITIONS = 'auto' | 'static' | 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+const POSITION: Record<string, POSITIONS> = {
+  AUTO: 'auto',
+  STATIC: 'static',
+  TOP_LEFT: 'top-left',
+  TOP_RIGHT: 'top-right',
+  BOTTOM_LEFT: 'bottom-left',
+  BOTTOM_RIGHT: 'bottom-right',
+};
+
+interface MenuProps extends Omit<BoxProps, 'children' | 'className'> {
+  /** If true, the menu will be active. */
+  active?: boolean;
+  /** The content to display inside the menu. */
+  children?: ReactNode;
+  /** A class name for the wrapper to give custom styles. */
+  className?: string;
+  /** Callback function that is fired when the menu hides. */
+  onHide?: () => void;
+  /** Callback function that is fired when a menu item is clicked. */
+  onSelect?: (e: SyntheticEvent) => void;
+  /** Callback function that is fired when the menu shows. */
+  onShow?: () => void;
+  /** If true, a border is rendered around the menu. */
+  outline?: boolean;
+  /** The position in which the menu is rendered. */
+  position?: POSITIONS;
+  /** If true, the menu will highlight the selected value. */
+  selectable?: boolean;
+  /** The value of the menu item that will be highlighted. */
+  selected?: any;
+}
+
+const Menu: GenericComponent<MenuProps> = ({
+  active = false,
+  children,
+  className,
+  onHide,
+  onSelect,
+  onShow,
+  outline = true,
+  position = 'static',
+  selectable = true,
+  selected,
+  ...others
+}) => {
+  const [stateWidth, setWidth] = useState<number | undefined>(0);
+  const [stateHeight, setHeight] = useState<number | undefined>(0);
+  const [stateActive, setActive] = useState<boolean>(active);
+  const [statePosition, setPosition] = useState<string | undefined>(position);
+
+  const menuNode = useRef<HTMLUListElement>(null);
+  const menuWrapper = useRef<HTMLElement>(null);
+
+  const boxProps = pickBoxProps(others);
+  const classNames = cx(
+    theme['menu'],
+    theme[position],
+    {
+      [theme['active']]: active,
+    },
+    className,
+  );
+  const outlineClassNames = cx(theme['outline'], {
+    [uiUtilities['box-shadow-200']]: position !== POSITION.STATIC,
+  });
+
+  const handleDocumentClick = (event: SyntheticEvent) => {
+    if (stateActive && !events.targetIsDescendant(event, menuWrapper.current)) {
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      hide();
+    }
+  };
+
+  const handleSelect = (item: ReactElement, event: SyntheticEvent) => {
+    const { value, onClick } = item.props;
+
+    if (onSelect) {
+      onSelect(value);
+    }
+
+    if (onClick) {
+      event.persist();
+      onClick(event);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    hide();
```

Comment:
why you dont move it up so its lint-okay? same for other occurrences

Reply:
Well that's the only simpliest way I found, there are chained functions. If you move `hide` up, then you have to move `removeEvents` as well. And then `removeEvents` has `handleDocumentClick` which calls `hide`. 😄